Add support for processing language parameter in message entities

pplulee opened the pull request:
## Description
Add support for processing code language parameter in `<pre>` tag under message entities

## Describe your tests
How did you test your change?

Python version: 3.12

OS: Windows 11

Send a message containing a code block to the bot, then see the result of the parsed HTML code.

![image](https://github.com/user-attachments/assets/3ba5aa32-4ea0-488f-b7bd-3b91bfc99c7d)
![image](https://github.com/user-attachments/assets/012d4b1a-5237-4867-b1f9-e6ac5c880fba)
## Checklist:
- [ ] I added/edited example on new feature/change (if exists)
- [x] My changes won't break backward compatibility
- [ ] I made changes both for sync and async

Badiboy commented:
Looks fine. Did you perform tests of this update?

pplulee commented:
I was unable to run some of the tests, but that seems to be the original problem, not caused by these updates.

For example, in `test_telebot.py`, I'm confused as to why empty data was used. This caused the tests to fail.
![image](https://github.com/user-attachments/assets/3f418734-ef96-4dec-aff6-78b5abdb1419)
![image](https://github.com/user-attachments/assets/fb2e1d88-eb9c-4057-9d1e-c5ca98b16879)

pplulee commented:
I updated the test file with some optimisations. For the issues mentioned above, environment variables are used directly. All tests now pass.

Also, I added tests for the `apply_html_entities` function.

Badiboy commented:
> For example, in `test_telebot.py`, I'm confused as to why empty data was used.

You need to pass YOUR bot token there.

Badiboy commented:
> For the issues mentioned above, environment variables are used directly.

Not absolutely correct due to TOKEN var may not exist if nothing in environment, but better then ''...

pplulee commented:
> > For example, in `test_telebot.py`, I'm confused as to why empty data was used.
> 
> You need to pass YOUR bot token there.

I saw environment variables for token, but some tests used them and some didn't, so I was confused. I updated to use the same environment variables for all tests so we don't have to manually edit the test files.

Badiboy commented:
func is designed wrong.

```
    def func(upd_text, subst_type=None, url=None, user=None, custom_emoji_id=None, language=None):
        upd_text = upd_text.decode("utf-16-le")
        if subst_type == "text_mention":
            subst_type = "text_link"
            url = "tg://user?id={0}".format(user.id)
        elif subst_type == "mention":
            url = "https://t.me/{0}".format(upd_text[1:])
        upd_text = upd_text.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;")
        if not subst_type or not _subs.get(subst_type):
            return upd_text
        subs = _subs.get(subst_type)
        if subst_type == "custom_emoji":
            return subs.format(text=upd_text, custom_emoji_id=custom_emoji_id)
        elif (subst_type == "pre") and language:
                return upd_text='<code class="language-{0}">{1}</code>'.format(language, upd_text)
        return subs.format(text=upd_text, url=url)

```
Otherwise symbols like < and > will break the markdown.

pplulee commented:
> func is designed wrong.
> 
> ```
>     def func(upd_text, subst_type=None, url=None, user=None, custom_emoji_id=None, language=None):
>         upd_text = upd_text.decode("utf-16-le")
>         if subst_type == "text_mention":
>             subst_type = "text_link"
>             url = "tg://user?id={0}".format(user.id)
>         elif subst_type == "mention":
>             url = "https://t.me/{0}".format(upd_text[1:])
>         upd_text = upd_text.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;")
>         if not subst_type or not _subs.get(subst_type):
>             return upd_text
>         subs = _subs.get(subst_type)
>         if subst_type == "custom_emoji":
>             return subs.format(text=upd_text, custom_emoji_id=custom_emoji_id)
>         elif (subst_type == "pre") and language:
>                 return upd_text='<code class="language-{0}">{1}</code>'.format(language, upd_text)
>         return subs.format(text=upd_text, url=url)
> ```
> 
> Otherwise symbols like < and > will break the markdown.

Thanks for pointing out the problem. I've modified the code along the lines of your idea and added a test for special characters.

Badiboy commented:
Thank you!
